funding/manager.go

```diff
@@ -2704,6 +2733,82 @@ func (f *Manager) waitForTimeout(completeChan *channeldb.OpenChannel,
 	}
 }
 
+// checkFundingInputsSpent checks whether the inputs to the funding transaction
+// are spent by another transaction. This will allow the funding manager to
+// forget the channel.
+func (f *Manager) checkFundingInputsSpent(c *channeldb.OpenChannel) error {
+	numInputs := len(c.FundingTxn.TxIn)
+	errChan := make(chan error, numInputs)
+	spendChan := make(chan *chainntnfs.SpendDetail, numInputs)
+	done := make(chan interface{})
+
+	for _, input := range c.FundingTxn.TxIn {
+		f.wg.Add(1)
+		go func(txIn *wire.TxIn) {
+			defer f.wg.Done()
+			// We use the zero-taproot-pk-script here since it will
+			// only notify on the outpoint being spent and not the
+			// outpoint+pkscript. This is because:
+			// - it's not necessary to be notified on the pkscript
+			//   being spent.
+			// - we cannot use ComputePkScript for an input that
+			//   spends a taproot output.
+			zeroScript := chainntnfs.ZeroTaprootPkScript.Script()
+			spendNtfn, err := f.cfg.Notifier.RegisterSpendNtfn(
+				&txIn.PreviousOutPoint, zeroScript,
+				c.BroadcastHeight()-fundingBroadcastBuffer,
+			)
+			if err != nil {
+				errChan <- err
+				return
+			}
+
+			defer spendNtfn.Cancel()
+
+			select {
+			case spend, ok := <-spendNtfn.Spend:
+				if !ok {
+					err := fmt.Errorf("spend chan closed")
+					errChan <- err
+					return
+				}
+
+				// Send along the spendChan.
+				spendChan <- spend
+
+			case <-f.quit:
+				// The funding manager is quitting.
+
+			case <-done:
+				// Another goroutine is cleaning us up.
+			}
+		}(input)
+	}
+
+	select {
+	case spend := <-spendChan:
+		// Close the done channel so that the other goroutines can clean
+		// up.
+		close(done)
```

Comment:
Since we block on getting one of these results, can't we just `defer close(done)` once at the top of this function?